website/docs/cdktf/create-and-deploy/deployment-patterns.mdx

```diff
@@ -0,0 +1,175 @@
+        with:
+          terraform_version: 1.1.7
+      - name: Install dependencies
+        run: npm ci
+      # If you use providers or modules that are not pre-built, you can install them here
+      - name: Generate bindings for providers and modules
+        run: cdktf get
+      - name: Run Tests
+        run: npm test
+      - name: Deploy
+        run: cdktf deploy --auto-approve '*' # Deploys all stacks
+        env:
+          TERRAFORM_CLOUD_TOKEN: ${{ secrets.TF_API_TOKEN }}
+```
+
+To get the plan output posted as a comment on your pull request, add the following to your `.github/workflows/cdktf-diff.yml` file:
+
+```yaml
+name: CDKTF Diff
+
+on: [pull_request]
+
+jobs:
+  diff:
+    name: "Terraform CDK Diff"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.1.7
+          cli_config_credentials_token: ${{ secrets.TF_CLOUD_TOKEN }}
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "14"
+
+      - run: yarn install
+
+      # We run the plan here, you need to fill in the name of your stack
+      - name: plan
+        run: cdktf plan <stack name>
+        env:
+          TERRAFORM_CLOUD_TOKEN: ${{ secrets.TF_API_TOKEN }}
+        continue-on-error: true # We want to see the plan output even if the plan fails
+
+      # You can replace this step with any workflow you like, e.g. posting a message in a slack channel
+      - uses: actions/github-script@0.9.0
+        if: github.event_name == 'pull_request'
+        env:
+          PLAN: "terraform\n${{ steps.plan.outputs.stdout }}" # We get the output from the last step
+        with:
+          github-token: ${{ secrets.GH_COMMENT_TOKEN }} # You need to create a secret with a github token that can comment on PRs
+          script: |
+            const output = `#### Terraform Plan 📖\`${{ steps.plan.outcome }}\`
+            <details><summary>Show Plan</summary>
+            \`\`\`${process.env.PLAN}\`\`\`
+            </details>
+            *Pusher: @${{ github.actor }}, Action: \`${{ github.event_name }}\`, Working Directory: \`${{ env.tf_actions_working_dir }}\`, Workflow: \`${{ github.workflow }}\`*`;
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: output
+            })
+```
+
+Please be aware that you need a job like this for every stack you have as you have to pass the stack name into `cdktf diff <stack name>`. None of the stacks can be dependent on another one since you can only reliably plan a stack if the stack(s) it depends on are already deployed in their latest version. This is also the reason why `cdktf diff` only support a single stack and not multiple like `cdktf deploy` or `cdktf destroy` do.
```

Comment:
I thin kthis should come before the examples so users can make an informed decision sooner about whether this workflow meets their use case.
```suggestion
```